lib/Doctrine/Common/Proxy/ProxyGenerator.php

```diff
@@ -1006,7 +1005,7 @@ private function buildParametersString(array $parameters)
                 $parameterDefinition .= '...';
             }
 
-            $parameterDefinition .= '$' . $param->getName();
+            $parameterDefinition .= '$' . (isset($renameParameters[$i]) ? $renameParameters[$i] : $param->getName());
```

Comment:
Why would you suggest to make it use $param->getName() when $i is 0?

Reply:
```suggestion
            $parameterDefinition .= '$' . ($renameParameters ? $renameParameters[$i] : $param->getName());
```

😛

Reply:
you could do 

```php
$parameterDefinition .= '$' . ($renameParameters[$i] ?? $param->getName());
```

if you want to make it fancy. The other two suggestions are buggy I think

Reply:
No, they just reveal bugs instead of hiding them. Intention here is to make this fail purposefully when you specify offsets that do not exist. This is called [Offensive programming](https://en.wikipedia.org/wiki/Offensive_programming).

Reply:
@ostrolucky well, if your intention is to enforce that `$renameParameters` array is always empty or the same size as `$params` it might be nicer to just check that before the loop and throw an exception if not. At least that would be easier to debug. But well, I guess this is all very much in bikeshedding territory :-)

Reply:
Yeah but that requires extra check and more code, while achieving same thing. This is basically Offensive programming + KISS/minimalism

Reply:
I'm just trying to imagine the failure mode: If `$renameParameters[$i]` is missing, it's probably only going to print a notice (that the dev may or may not notice) and then produce a function signature that looks like `function __set($name, $)` which gets written to the proxy file. And when the proxy file is loaded later on it'll trigger a parse error. Not really obvious that the parse error comes from some wrong array size somewhere else.

But like I said, it's a private method, so this is really mostly bikeshedding. As long as generation is properly covered by unittests it should be fine either way